src/sys/windows/stdnet/stream.rs

```diff
@@ -373,31 +373,25 @@ impl IntoRawSocket for UnixStream {
 struct TempPath(PathBuf);
 
 fn sample_ascii_string(len: usize) -> io::Result<String> {
-    const RANGE: u32 = 26 + 26 + 10;
     const GEN_ASCII_STR_CHARSET: &[u8] = b"ABCDEFGHIJKLMNOPQRSTUVWXYZ\
             abcdefghijklmnopqrstuvwxyz\
-            0123456789";
+            0123456789-_";
     let mut result = String::with_capacity(len);
+    let mut buf = [0; 4];
     for _ in 0..len {
```

Comment:
Is the loop still necessary? Can we make a single call to fill the entire string in one go?

Reply:
removed this loop! in practice now it should only be a single call but I added a new loop to avoid overflow and make sure this function is safe even for really long strings